### PR TITLE
Add class for constraining bonds with unit tests

### PR DESCRIPTION
**Pull request by rsdefever**
Draft PR for class around bond constraints. Partially addresses #1.

### CODE REVIEW

**Comment by adowling2** [comment]
Do you want me to review?

**Comment by rsdefever** [comment]
@adowling2 yes I think this is ready for review. Watch out for my `force-push`...didn't realize you were looking this branch quite yet.

**Comment by rsdefever** [comment]
@adowling2 I added install instructions to the README in this PR so you can run the tests as well.

File: bondconstrain/constrainmol.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        # Extract coordinates and bonds
+        xyz = structure.coordinates
+        constraints = {}
+        for bond in structure.bonds:
+            idx1 = bond.atom1.idx
+            idx2 = bond.atom2.idx
+            constraints[(idx1, idx2)] = bond.type.req
+
+        self.structure = deepcopy(structure)
+        self.model = self._create_model(xyz, constraints)
```
Comment (adowling2) [inline bondconstrain/constrainmol.py:40]: Add `self.model_solved = False`.

File: bondconstrain/constrainmol.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        )
+        if not success:
+            raise ValueError("Optimal solution not found.")
+        constrained_xyz = np.zeros((len(self.model.atom_ids), 3))
+        for idx in self.model.atom_ids:
+            constrained_xyz[idx, 0] = self.model.x[idx].value
+            constrained_xyz[idx, 1] = self.model.y[idx].value
+            constrained_xyz[idx, 2] = self.model.z[idx].value
+
+        self.structure.coordinates = constrained_xyz
```
Comment (adowling2) [inline bondconstrain/constrainmol.py:164]: Add `self.model_solved = True`

File: bondconstrain/constrainmol.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+                f"{self.structure.coordinates.shape}."
+            )
+        self.structure.coordinates = xyz
+        for idx in range(xyz.shape[0]):
+            self.model.x_start[idx] = xyz[idx, 0]
+            self.model.y_start[idx] = xyz[idx, 1]
+            self.model.z_start[idx] = xyz[idx, 2]
+            self.model.x[idx] = xyz[idx, 0]
+            self.model.y[idx] = xyz[idx, 1]
+            self.model.z[idx] = xyz[idx, 2]
```
Comment (adowling2) [inline bondconstrain/constrainmol.py:198]: Add `self.model_solved = False`. This way you are keeping track of when you update the model but did not resolve yet.

**Review by adowling2 — APPROVED** [review]
Looks good to me. I did not try running it.

**Pull request merged**